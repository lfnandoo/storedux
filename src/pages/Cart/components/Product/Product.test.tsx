import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

import { Product } from "./index";

describe("Product", () => {
  const onDecrementProduct = jest.fn();
  const onIncrementProduct = jest.fn();
  const onRemoveProduct = jest.fn();
  const item = {
    product: {
      id: "1",
      name: "Luiz",
      price: "350.32",
      image: "/",
    },
    amount: 2,
  };
  const component = (
    <Product
      item={item}
      onDecrementProduct={onDecrementProduct}
      onIncrementProduct={onIncrementProduct}
      onRemoveProduct={onRemoveProduct}
    />
  );

  it("should render correctly", () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should calc price", () => {
    const { getByTestId } = render(component);

    expect(getByTestId("formated-price")).toHaveTextContent("R$ 700,64");
  });

  it("should trigger onIncrementProduct", () => {
    const { getByTestId } = render(component);

    fireEvent.click(getByTestId("increment-btn"));

    expect(onIncrementProduct).toHaveBeenCalled();
  });

  it("should trigger onDecrementProduct", () => {
    const { getByTestId } = render(component);

    fireEvent.click(getByTestId("decrement-btn"));

    expect(onDecrementProduct).toHaveBeenCalled();
  });

  it("should trigger onRemoveProduct", () => {
    const { getByTestId } = render(
      <Product
        item={{
          product: {
            id: "1",
            name: "Luiz",
            price: "350.32",
            image: "/",
          },
          amount: 1,
        }}
        onDecrementProduct={onDecrementProduct}
        onIncrementProduct={onIncrementProduct}
        onRemoveProduct={onRemoveProduct}
      />
    );

    fireEvent.click(getByTestId("decrement-btn"));

    expect(onRemoveProduct).toHaveBeenCalled();
  });
});
