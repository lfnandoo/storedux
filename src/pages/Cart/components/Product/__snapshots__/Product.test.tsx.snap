// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should render correctly 1`] = `
<div
  className="sc-bdnxRM sqYLY"
>
  <img
    alt="Luiz"
    src="/"
  />
  <div
    className="sc-gtsrHT dfOuWm"
  >
    <div
      className="sc-dlnjwi VCOiA"
    >
      <span>
        Luiz
      </span>
      <p>
        R$ 350,32
         x 
        2
         =
         
        <strong
          data-testid="formated-price"
        >
          R$ 700,64
        </strong>
      </p>
    </div>
    <div
      className="sc-hKFxyN bbrNVV"
    >
      <p
        data-testid="decrement-btn"
        onClick={[Function]}
      >
        -
      </p>
      <span>
        2
      </span>
      <p
        data-testid="increment-btn"
        onClick={[Function]}
      >
        +
      </p>
    </div>
  </div>
</div>
`;
